### ISSUE TITLE
Extension causes high cpu load

### ISSUE THREAD

**Issue by shurasanchos** [shd101wyy/vscode-markdown-preview-enhanced#228]
- Issue Type: `Performance`
- Extension Name: `markdown-preview-enhanced`
- Extension Version: `0.3.13`
- OS Version: `Windows_NT x64 10.0.17763`
- VSCode version: `1.34.0`

:warning: Make sure to **attach** this file from your *home*-directory:
:warning:`C:\Users\pupsi\shd101wyy.markdown-preview-enhanced-unresponsive.cpuprofile.txt`

Find more details here: https://github.com/Microsoft/vscode/wiki/Explain:-extension-causes-high-cpu-load

**Comment by shd101wyy** [comment]
Yeah I also encountered the issue when I was debugging the extension. 
But I couldn't find why it happens. 
I would need more time to dig into it. Thanks.

**Comment by crystalfp** [comment]
As requested, here is the log file from the extension.
I don't understand why it is activated when the files are not markdown.
[shd101wyy.markdown-preview-enhanced-unresponsive.cpuprofile.txt.zip](https://github.com/shd101wyy/vscode-markdown-preview-enhanced/files/3972389/shd101wyy.markdown-preview-enhanced-unresponsive.cpuprofile.txt.zip)

**Comment by jdavies** [comment]
Just started seeing this ye
[shd101wyy.markdown-preview-enhanced-unresponsive.cpuprofile.txt](https://github.com/shd101wyy/vscode-markdown-preview-enhanced/files/4193256/shd101wyy.markdown-preview-enhanced-unresponsive.cpuprofile.txt)
sterday.

**Comment by showa-yojyo** [comment]
To the developer of this extension:

[shd101wyy.markdown-preview-enhanced-unresponsive.cpuprofile.txt](https://github.com/shd101wyy/vscode-markdown-preview-enhanced/files/4253367/shd101wyy.markdown-preview-enhanced-unresponsive.cpuprofile.txt)

**Comment by crystalfp** [comment]
Here is mine most recent one:
[shd101wyy.markdown-preview-enhanced-unresponsive.cpuprofile.txt](https://github.com/shd101wyy/vscode-markdown-preview-enhanced/files/4490971/shd101wyy.markdown-preview-enhanced-unresponsive.cpuprofile.txt)

**Comment by liuxiang0** [comment]
Here  is mine
[shd101wyy.markdown-preview-enhanced-unresponsive.cpuprofile.txt](https://github.com/shd101wyy/vscode-markdown-preview-enhanced/files/4810856/shd101wyy.markdown-preview-enhanced-unresponsive.cpuprofile.txt)

VSCODE Version: 1.46.1
Date: 2020-06-17T21:13:08.304Z
Electron: 7.3.1
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Linux x64 5.3.0-59-generic

**Comment by thaenalpha** [comment]
Happen to me too, here is mine
[shd101wyy.markdown-preview-enhanced-unresponsive.cpuprofile.txt](https://github.com/shd101wyy/vscode-markdown-preview-enhanced/files/4936160/shd101wyy.markdown-preview-enhanced-unresponsive.cpuprofile.txt)

Version: 1.48.0-insider (user setup)
Commit: 5e5fa3edde19d38ac512d65ccefe2a12ecab025a
Date: 2020-07-16T13:12:57.510Z
Electron: 8.3.3
Chrome: 80.0.3987.165
Node.js: 12.13.0
V8: 8.0.426.27-electron.0
OS: Windows_NT x64 10.0.20161

**Comment by appleobject** [comment]
Happen to me too, here is mine 
[shd101wyy.markdown-preview-enhanced-unresponsive.cpuprofile.txt](https://github.com/shd101wyy/vscode-markdown-preview-enhanced/files/4943541/shd101wyy.markdown-preview-enhanced-unresponsive.cpuprofile.txt)

**Comment by SteveGilham** [comment]
Attaching this, like it asked me to -- 
[shd101wyy.markdown-preview-enhanced-unresponsive.cpuprofile.txt](https://github.com/shd101wyy/vscode-markdown-preview-enhanced/files/5098132/shd101wyy.markdown-preview-enhanced-unresponsive.cpuprofile.txt)

**Comment by aunraza** [comment]
[shd101wyy.markdown-preview-enhanced-unresponsive.cpuprofile.txt](https://github.com/shd101wyy/vscode-markdown-preview-enhanced/files/5147097/shd101wyy.markdown-preview-enhanced-unresponsive.cpuprofile.txt)

**Comment by GitHubCloud** [comment]
[shd101wyy.markdown-preview-enhanced-unresponsive.cpuprofile.txt](https://github.com/shd101wyy/vscode-markdown-preview-enhanced/files/5229822/shd101wyy.markdown-preview-enhanced-unresponsive.cpuprofile.txt)

**Comment by r1j1k** [comment]
Issue is opened since May 2019... crap
[shd101wyy.markdown-preview-enhanced-unresponsive.cpuprofile.txt](https://github.com/shd101wyy/vscode-markdown-preview-enhanced/files/5265997/shd101wyy.markdown-preview-enhanced-unresponsive.cpuprofile.txt)
Any use?

**Comment by glodov** [comment]
I have the same issue 200%+ on macOS Big Sur.

**Comment by tillig** [comment]
I looked at a few of the profiles and the only thing I see in common is that when the high CPU load is hit, it's in the middle of loading a ton of `node_modules` into memory. All of the line numbers specified are 0, and there seems to be a lot of reference to `node_modules` locations. In the extension code I noticed there's a lot of `import *` which, based on the engine, will just recursively load _everything_ instead of only what's used. And I didn't see in any build/package script here any notion of webpacking the extension to trim out stuff that the extension (or referenced libraries) won't use at runtime. [The MS extension guidelines even mention "loading 100 small files is much slower than loading one large one."](https://code.visualstudio.com/api/working-with-extensions/bundling-extension)

If I had to guess - I'd say that if the extension was webpacked as part of distribution and the extra stuff was excluded, this unresponsive startup thing could be alleviated, at least a bit.

**Comment by glodov** [comment]
Hello!
It was fixed on my side by uninstalling some of HTML/CSS extensions.
For some people uninstalling eslint helped (read from other solved tickets).

**Comment by fakhamatia** [comment]
[shd101wyy.markdown-preview-enhanced-unresponsive.cpuprofile.txt](https://github.com/shd101wyy/vscode-markdown-preview-enhanced/files/6584390/shd101wyy.markdown-preview-enhanced-unresponsive.cpuprofile.txt)

**Comment by Clearlove7788** [comment]
It works well, but it seems to be a bit slow

**Comment by NKarolak** [comment]
I have no HTML/CSS or eslint extension installed.
But still, the extension did not respond soon enough, as per VS Code notifications.

[shd101wyy.markdown-preview-enhanced-unresponsive.cpuprofile.txt](https://github.com/shd101wyy/vscode-markdown-preview-enhanced/files/6738862/shd101wyy.markdown-preview-enhanced-unresponsive.cpuprofile.txt)

**Comment by germanolira** [comment]
I have the same issue

**Comment by sospiess** [comment]
Showing up for the first time after using the extension for a couple of months. -- After disabling some other extensions at the workspace level to speed up activation time. ^_^

Nonetheless, thanks for the great extension! =)
[dendron.dendron-markdown-preview-enhanced-unresponsive.cpuprofile.txt](https://github.com/shd101wyy/vscode-markdown-preview-enhanced/files/9598033/dendron.dendron-markdown-preview-enhanced-unresponsive.cpuprofile.txt)
